### ISSUE TITLE
Make ‘Ran’, ‘Lan’ kind polymorphic

### ISSUE THREAD

**Issue by Icelandjack** [ekmett/kan-extensions#31]
```haskell
data Lan :: (k -> Type) -> (k -> Type) -> (Type -> Type) where
  Lan :: (g b -> a) -> (h b) -> Lan g h a

newtype Ran :: (k -> Type) -> (k -> Type) -> (Type -> Type) where
  Ran :: { runRan :: forall b. (a -> g b) -> h b } -> Ran g h a
```

**Comment by RyanGlScott** [comment]
Sounds reasonable. Pull requests welcome.

**Comment by RyanGlScott** [comment]
Fixed in #40.
